src/Board.java

```diff
@@ -97,4 +106,40 @@ private void updatePieceCount(int position) {
             this.numberOfBlackPieces--;
         }
     }
+    
+    public boolean isEndState() {
+        return (numberOfBlackPieces == 0 || numberOfWhitePieces == 0);
+    }
+
+    public List<Square> getAdjacentSquares(Square square) {
+        List<Integer> squareNumbers = square.getAdjacentSquares();
+        return getSquares((ArrayList<Integer>) squareNumbers);
+    }
+
+    public List<Square> getSquaresTwoMovesAway(Square square) {
```

Comment:
Could you rename the parameter to `startingSquare` please? I think that helps make the code a bit more readable given the nested for loops.